tests/acceptance/features/bootstrap/FeatureContext.php

```diff
@@ -1206,13 +1220,13 @@ public function userHasSentHTTPMethodToUrlWithPassword(string $user, string $ver
 	 * @param string|null $user
 	 * @param string|null $verb
 	 * @param string|null $url
-	 * @param TableNode|null $body
+	 * @param TableNode|string|null $body
 	 * @param string|null $password
 	 *
 	 * @return void
 	 * @throws GuzzleException
 	 */
-	public function sendingToWithDirectUrl(?string $user, ?string $verb, ?string $url, ?TableNode $body, ?string $password = null): void {
+	public function sendingToWithDirectUrl(?string $user, ?string $verb, ?string $url, $body = null, ?string $password = null): void {
```

Comment:
Could you do this in this function and do setResponse() from When steps only and and if used in Given do the required assertion there 
```diff
- $this->response = HttpRequestHelper::sendRequest(
-	$fullUrl,
-	$this->getStepLineRef(),
-	$verb,
-	$user,
-	$password,
-	$headers,
-	$bodyRows,
-	$config,
-	$cookies
- );
+ return HttpRequestHelper::sendRequest(
+	$fullUrl,
+	$this->getStepLineRef(),
+	$verb,
+	$user,
+	$password,
+	$headers,
+	$bodyRows,
+	$config,
+	$cookies
+ );
```